scripts/pylib/twister/harness.py

```diff
@@ -160,7 +179,8 @@ def handle(self, line):
             is writen into handler.log
         '''
         self.state = "passed"
-        self.tests[self.id] = "PASS"
+        tc = self.instance.get_case_or_create(self.id)
```

Comment:
The self.id here is a testsuite name, so a new testcase which name is as same as "testsuite name" will be created each time. This is a corner case.

Reply:
It is only affect the tests/subsys/debug/coredump_backend/.  I tried some thing on this:
```
diff --git a/scripts/pylib/twister/harness.py b/scripts/pylib/twister/harness.py
index 2c7f076b25..07694431a4 100644
--- a/scripts/pylib/twister/harness.py
+++ b/scripts/pylib/twister/harness.py
@@ -53,6 +53,7 @@ class Harness:
         self.instance = None
         self.testcase_output = ""
         self._match = False
+        self.curr_tc_name = None

     def configure(self, instance):
         self.instance = instance
@@ -97,6 +98,7 @@ class Harness:
             self.capture_coverage = False

 class Console(Harness):
+    ZTEST_START_PATTERN = r"START - (test_)?(.*)"

     def configure(self, instance):
         super(Console, self).configure(instance)
@@ -108,6 +110,10 @@ class Console(Harness):
                 self.patterns.append(re.compile(r))

     def handle(self, line):
+        tc_match = re.search(self.ZTEST_START_PATTERN, line)
+        if tc_match:
+            self.curr_tc_name = "{}.{}".format(self.id, tc_match.group(2))
+
         if self.type == "one_line":
             if self.pattern.search(line):
                 self.state = "passed"
@@ -152,7 +158,10 @@ class Console(Harness):

         self.process_test(line)

-        tc = self.instance.get_case_or_create(self.id)
+        tc = self.instance.get_case_by_name(self.curr_tc_name)
+        if not tc:
+            return
+
         if self.state == "passed":
             tc.status = "passed"
         else:
@@ -272,7 +281,9 @@ class Test(Harness):

         if match and match.group(2):
             name = "{}.{}".format(self.id, match.group(3))
-            tc = self.instance.get_case_or_create(name)
+            tc = self.instance.get_case_by_name(name)
+            if not tc:
+                return

             matched_status = match.group(1)
             tc.status = self.ztest_to_status[matched_status]
```

But I think this is a very corner case and the testcase should be fixed.